app/controllers/carts_controller.rb

```diff
@@ -1,25 +1,27 @@
 class CartsController < ApplicationController
   def create
-    if session[:cart]
-      if session[:cart][params[:item_id]]
-        session[:cart][params[:item_id]]  += 1
-      else
-        session[:cart][params[:item_id]] = 1
-      end
-    else
-      session[:cart] = {params[:item_id] => 1}
-    end
+    session[:cart] ||= {}
+    session[:cart][params[:item_id]] ||= 0
+    session[:cart][params[:item_id]] += 1
     @cart_items = session[:cart]
 
-    flash[:notice] = 'Added '  + Item.find_by_dao(params[:item_id]).name
+    flash[:notice] = 'Added ' + Item.find_by_dao(params[:item_id]).name
     redirect_to '/cart'
   end
 
   def show
-    if session[:cart]
-      @cart_items = session[:cart]
-    else
-      @cart_items = ["No Cart Items"]
+    @cart_items = if session[:cart]
+      session[:cart].map do |item_dao, quantity|
+        CartItem.new(item_dao, quantity)
+      end
     end
+    @cart_total = grand_total
   end
+
+  private
+
+  def grand_total
+    @cart_items.reduce(0){|sum, cart_item| sum += cart_item.total}
```

Comment:
Useless assignment to variable - `sum`. Use just operator `+`.<br>Prefer `inject` over `reduce`.<br>Space missing to the left of {.<br>Space between { and | missing.<br>Space missing inside }.